qa/integration-tests/src/test/java/io/camunda/zeebe/it/clustering/ClusteredSnapshotTest.java

```diff
@@ -11,85 +11,68 @@
 import io.camunda.zeebe.protocol.record.Record;
 import io.camunda.zeebe.protocol.record.RecordType;
 import io.camunda.zeebe.protocol.record.ValueType;
-import io.camunda.zeebe.snapshots.SnapshotId;
 import java.time.Duration;
 import java.util.Collections;
-import java.util.Map;
-import java.util.concurrent.atomic.AtomicLong;
-import java.util.concurrent.atomic.AtomicReference;
 import java.util.function.Consumer;
-import java.util.function.Predicate;
-import java.util.stream.Collectors;
 import java.util.stream.IntStream;
+import java.util.stream.Stream;
 import org.assertj.core.api.AbstractAssert;
 import org.assertj.core.api.Assertions;
 import org.awaitility.Awaitility;
-import org.junit.After;
-import org.junit.Rule;
-import org.junit.Test;
-import org.junit.runner.RunWith;
-import org.junit.runners.Parameterized;
-import org.junit.runners.Parameterized.Parameter;
-import org.junit.runners.Parameterized.Parameters;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Named;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.util.unit.DataSize;
 
-@RunWith(Parameterized.class)
-public class ClusteredSnapshotTest {
+final class ClusteredSnapshotTest {
 
   public static final Logger LOG = LoggerFactory.getLogger("ClusteredSnapshotTest");
   private static final Duration SNAPSHOT_INTERVAL = Duration.ofMinutes(5);
 
-  @Rule
-  public final ClusteringRule clusteringRule = new ClusteringRule(1, 3, 3, this::configureBroker);
-
-  @Parameter(0)
-  public Consumer<ClusteringRule> snapshotTrigger;
-
-  @Parameter(1)
-  public String description;
-
-  @Parameters(name = "{index}: {1}")
-  public static Object[][] snapshotTriggers() {
-    return new Object[][] {
-      new Object[] {
-        (Consumer<ClusteringRule>)
-            (rule) -> {
-              LOG.info("Triggerring snapshots using admin api");
-              rule.triggerAndWaitForSnapshots();
-            },
-        "explicit trigger snapshot"
-      },
-      new Object[] {
-        (Consumer<ClusteringRule>)
-            (rule) -> {
-              LOG.info("Increasing clock by snapshot interval {}", SNAPSHOT_INTERVAL);
-              rule.getClock().addTime(SNAPSHOT_INTERVAL);
-            },
-        "implicit snapshot by advancing the clock"
-      }
-    };
-  }
-
-  @After
-  public void cleanUp() {
+  @RegisterExtension
+  private final ClusteringRuleExtension clusteringRule =
+      new ClusteringRuleExtension(1, 3, 3, this::configureBroker);
+
+  public static Stream<Arguments> snapshotTriggers() {
+    return Stream.of(
+        Arguments.of(
+            Named.of(
+                "explicit trigger snapshot",
+                (Consumer<ClusteringRule>)
```

Comment:
:upside_down_face: I'd just make it a type itself - a type can implement `Named` as well, so e.g.

```java
private final record TestCase(String name, Consumer<ClusteringRule> snapshotTrigger) implements Named<Consumer<ClusteringRule>> {
  public Consumer<ClusteringRule> getPayload() {
    return snapshotTrigger;
  }

  public String getName() {
    return name;
  }
}

private static Stream<TestCase> snapshotTriggers() {
  return Stream.of(
    new TestCase("explicit", rule -> rule.triggerAndWaitForSnapshots()),
    new TestCase("implicit via clock", rule -> rule.getClock().addTime(SNAPSHOT_INTERVAL))
  );
}
```

I find it more readable, and more extensible in the future. But it's a very soft suggestion, as the impact is quite low :)